### PR TITLE
Conveyor update

### PR DESCRIPTION
**Pull request by Citinited**
I had some time on my hands recently, so decided to update conveyors. If you've ever made a conveyor belt that was longer than a short stretch, you'll know how infuriating they are to get right. I aimed to fix those issues in this PR.

List of changes:
- New sprites! These sprites make the direction of the conveyor belt really obvious, which means they are very, very easy to orient correctly.
- Conveyor belts will now automatically create corner pieces for you (see the .mp4 below).
- You can click a belt directly with a lever to link them.
- Clicking the ground with a conveyor belt construct will spawn a conveyor that faces your direction. This also applies to the conveyor belt placer.
- Removed levers will keep their ID instead of assigning a new one.
- Removed the "broken" sprite - AFAIK nothing makes conveyors broken, so I didn't bother with spriting these.
- Unpowered conveyor levers will no longer have a lit light.
- Loose conveyor switches have had their sprite slightly changed to help differentiate them from actual levers. (They looked exactly the same before)

I went through and refactored a bunch of this code. In doing so, I had to mapedit some things out, which is why I touched a few map files.

I've left all of the old sprites in the .dmi file, in case anyone wants to use them in future.

**Pictures!**

Loose lever and loose belt: image

A .mp4 of how the new sprites look can be found here: https://imgur.com/a/ZPuQIzg

A .mp4 of improved belt rotation: https://i.imgur.com/ZvoQ2d9.mp4

:cl:
tweak: Conveyor belts have received a facelift!
tweak: Rotating a belt with a wrench now nicely rotates the belt for you
tweak: Click a lever with a lever to link their IDs. Click a belt with a lever to link that belt to that lever.
/:cl:

This almost certainly can be improved, so reviews are (as always) appreciated.

### CODE REVIEW

**Comment by Citinited** [comment]
Damn you Travis for reminding me that Metastation is a thing

I need to go through the other station files to make sure that mapedited vars have been properly dealt with, but that'll be tomorrow

File: _maps/map_files/cyberiad/cyberiad.dmm [hunk 1]
```diff
@@ -5062,7 +5062,7 @@
 "bTr" = (/obj/machinery/door/firedoor,/obj/machinery/door/airlock/external{id_tag = "supply_home"; name = "Cargo Docking Hatch"; req_access_txt = "31"},/turf/simulated/floor/plating,/area/quartermaster/storage)
 "bTs" = (/turf/simulated/floor/plasteel{icon_state = "delivery"},/area/quartermaster/storage)
 "bTt" = (/obj/machinery/door/airlock/external{id_tag = "s_docking_airlock"; name = "Shuttle Hatch"; req_access_txt = "31"},/turf/simulated/shuttle/floor,/area/shuttle/supply)
-"bTu" = (/obj/machinery/conveyor{dir = 4; id = "QMLoad"},/obj/machinery/status_display/supply_display{pixel_y = -32},/obj/effect/decal/warning_stripes/north,/turf/simulated/floor/plating,/area/quartermaster/storage)
+"bTu" = (/obj/structure/plasticflaps/mining,/obj/machinery/conveyor{dir = 8; step_x = 0; step_y = 0; id = "QMLoad"},/turf/simulated/floor/plating,/area/quartermaster/storage)
```
Comment (alexkar598) [inline _maps/map_files/cyberiad/cyberiad.dmm]: Remove all steps from the map

File: _maps/map_files/cyberiad/cyberiad.dmm [hunk 2]
```diff
@@ -5100,7 +5100,7 @@
 "bUd" = (/obj/structure/window/reinforced,/turf/simulated/floor/plasteel{dir = 1},/area/medical/genetics)
 "bUe" = (/turf/simulated/floor/plasteel{tag = "icon-whiteblue (WEST)"; icon_state = "whiteblue"; dir = 8},/area/medical/medbay2)
 "bUf" = (/obj/machinery/door/window/southright{dir = 2; name = "Primate Pen"; req_access_txt = "9"},/turf/simulated/floor/plasteel{dir = 1},/area/medical/genetics)
-"bUg" = (/obj/machinery/conveyor{dir = 4; id = "QMLoad"},/obj/effect/decal/warning_stripes/north,/turf/simulated/floor/plating,/area/quartermaster/storage)
+"bUg" = (/obj/structure/plasticflaps/mining,/obj/machinery/conveyor{dir = 8; step_x = 0; step_y = 0; id = "QMLoad"},/turf/simulated/shuttle/plating,/area/shuttle/supply)
```
Comment (alexkar598) [inline _maps/map_files/cyberiad/cyberiad.dmm]: :eyes: step

File: _maps/map_files/cyberiad/cyberiad.dmm [hunk 3]
```diff
@@ -5240,7 +5240,7 @@
 "bWN" = (/obj/structure/dispenser,/turf/simulated/shuttle/floor{icon_state = "floor4"},/area/shuttle/administration)
 "bWO" = (/obj/structure/shuttle/engine/propulsion{dir = 8; icon_state = "propulsion_l"; tag = "icon-propulsion_l (EAST)"},/turf/space,/area/shuttle/administration)
 "bWP" = (/obj/machinery/door/airlock/external{id_tag = "s_docking_airlock"; name = "Shuttle Hatch"; req_access_txt = "31"},/obj/docking_port/mobile/supply,/obj/docking_port/stationary{dir = 8; dwidth = 5; height = 7; id = "supply_home"; name = "supply bay"; width = 12},/turf/simulated/shuttle/floor,/area/shuttle/supply)
-"bWQ" = (/obj/machinery/conveyor_switch/oneway{convdir = -1; id = "QMLoad"},/turf/simulated/floor/plasteel,/area/quartermaster/storage)
+"bWQ" = (/obj/machinery/door/poddoor{id_tag = "QMLoaddoor"; name = "supply dock loading door"},/obj/machinery/conveyor{dir = 8; step_x = 0; step_y = 0; id = "QMLoad"},/turf/simulated/floor/plating,/area/quartermaster/storage)
```
Comment (alexkar598) [inline _maps/map_files/cyberiad/cyberiad.dmm]: :eyes: step

File: _maps/map_files/cyberiad/cyberiad.dmm [hunk 4]
```diff
@@ -5324,9 +5324,9 @@
 "bYt" = (/obj/machinery/door/airlock/public/glass,/turf/simulated/shuttle/floor{icon_state = "floor4"},/area/shuttle/administration)
 "bYu" = (/obj/structure/shuttle/engine/propulsion{dir = 8; icon_state = "propulsion_r"; tag = "icon-propulsion_r (EAST)"},/turf/space,/area/shuttle/administration)
 "bYv" = (/turf/space,/turf/simulated/shuttle/wall{dir = 4; icon_state = "diagonalWall3"},/area/shuttle/administration)
-"bYw" = (/obj/machinery/conveyor{dir = 4; id = "QMLoad"},/obj/structure/plasticflaps/mining,/turf/simulated/floor/plating,/area/quartermaster/storage)
-"bYx" = (/obj/machinery/conveyor{dir = 4; id = "QMLoad"},/obj/structure/plasticflaps/mining,/turf/simulated/shuttle/plating,/area/shuttle/supply)
-"bYy" = (/obj/machinery/conveyor{dir = 4; id = "QMLoad"},/obj/machinery/door/poddoor{id_tag = "QMLoaddoor"; name = "supply dock loading door"},/turf/simulated/floor/plating,/area/quartermaster/storage)
+"bYw" = (/obj/machinery/light,/obj/item/radio/intercom{name = "station intercom (General)"; pixel_y = -28},/obj/effect/decal/warning_stripes/north,/obj/machinery/conveyor{dir = 8; step_x = 0; step_y = 0; id = "QMLoad"},/turf/simulated/floor/plating,/area/quartermaster/storage)
```
Comment (alexkar598) [inline _maps/map_files/cyberiad/cyberiad.dmm]: :eyes: step

File: _maps/map_files/cyberiad/cyberiad.dmm [hunk 5]
```diff
@@ -5324,9 +5324,9 @@
...
+"bYx" = (/obj/machinery/status_display/supply_display{pixel_y = -32},/obj/effect/decal/warning_stripes/north,/obj/machinery/conveyor{dir = 8; step_x = 0; step_y = 0; id = "QMLoad"},/turf/simulated/floor/plating,/area/quartermaster/storage)
```
Comment (alexkar598) [inline _maps/map_files/cyberiad/cyberiad.dmm]: :eyes: step

File: _maps/map_files/cyberiad/cyberiad.dmm [hunk 6]
```diff
@@ -5324,9 +5324,9 @@
...
+"bYy" = (/obj/effect/decal/warning_stripes/north,/obj/machinery/conveyor{dir = 8; step_x = 0; step_y = 0; id = "QMLoad"},/turf/simulated/floor/plating,/area/quartermaster/storage)
```
Comment (alexkar598) [inline _maps/map_files/cyberiad/cyberiad.dmm]: :eyes: step

**Comment by Citinited** [comment]
@alexkar598 I have taken steps to address your issues

Metastation has also been updated

**Comment by Citinited** [comment]
I've reworked belt rotation - now when you click a conveyor belt with a wrench, it will first rotate 90 degrees clockwise, then figure out what it should connect to, then update the belt accordingly!

(See OP for a video of that in action)

**Comment by Dinarzad** [comment]
Dem sweet ass Factorio style conveyor belts.
8/10 would drastically improve my life with again.

**Comment by Citinited** [comment]
@Crazylemon64 @Fox-McCloud I thought of something - do you want the treadmill icons to be updated to the new sprites?

**Comment by taukausanake** [comment]
Am I the only one who doesn't really like the new sprites? Maybe. I like the old ones more but this does make it easier to know which way they are facing

**Comment by Citinited** [comment]
Personal preference really, but the older ones are much more muted while the newer ones are much more pronounced.

Obviously I prefer the newer ones 😛

File: code/modules/recycling/conveyor2.dm [hunk 7]
```diff
@@ -1,388 +1,413 @@
...
+	if(iscrowbar(I))
+		var/obj/item/conveyor_switch_construct/C = new(loc, id)
 		transfer_fingerprints_to(C)
-		to_chat(user,"<span class='notice'>You deattach the conveyor switch.</span>")
+		to_chat(user,"<span class='notice'>You detach the conveyor switch.</span>")
 		qdel(src)
-
-	else if(istype(I, /obj/item/multitool))
+	else if(ismultitool(I))
 		update_multitool_menu(user)
```
Comment (Citinited) [inline code/modules/recycling/conveyor2.dm]: I'll probably just change this to a `You set the one-way lever to two-way`.

**Comment by Citinited** [comment]
Going to squash those conflicts

**Comment by Citinited** [comment]
Now that Deltastation is a thing, I'll be updating it with the new conveyors

**Comment by variableundefined** [comment]
@Citinited This PR has been approved by a head and maintainer. Please inform us when you are ready to fix conflicts and update Deltastation, then we could work together to get this merged.

**Comment by Citinited** [comment]
About to fix conflicts and update delta. Will be ready in ~1 hour

**Comment by Citinited** [comment]
@variableundefined All conflicts fixed and delta added, let me know if there are any changes that need to be made (assuming travis doesn't fail)

**Pull request merged**